internal/http.go

```diff
@@ -0,0 +1,239 @@
+package internal
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/operate-first/curator-operator/pkg/signals"
+	"github.com/spf13/viper"
+
+	"github.com/jackc/pgx/v4"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// Server defines the http server configuration
+type Server struct {
+	Addr        string
+	ErrorLog    *log.Logger
+	BaseContext func(net.Listener) context.Context
```

Comment:
I think we'll need to drop any internal field that contains a reference to a context.Context. Any method/function that requires a context.Context should accept it as a parameter so the call site can handle any context-sensitive handelement as needed.